src/org/zaproxy/zap/extension/bruteforce/BruteForceParam.java

```diff
@@ -164,13 +180,73 @@ public void setFileExtensions(String fileExtensions) {
 	    }
 	    
 	    List<String> fileExtensionsList = new ArrayList<>();
-	    for (String fileExtension: fileExtensions.replaceAll("\\s", 
-	    		EMPTY_STRING).split(",")) {
+	    for (String fileExtension: fileExtensions.replaceAll(WHITESPACE, 
+	    		EMPTY_STRING).split(COMMA)) {
 	    	if (!fileExtension.equals(EMPTY_STRING)) {
 	    		fileExtensionsList.add(fileExtension);
 	    	}
 	    }
 	    
     	return fileExtensionsList;	    
 	}
+
+	/**
+	 * @return {@code String} of comma-separated file-extensions that are not to be processed.
+	 * {@code "jpg, gif, jpeg, ico, tiff, png, bmp"} is returned by default
+	 */
+	public String getExtensionsToMiss() {
+		return extensionsToMiss;
+	}
+
+	/**
+	 * Define a {@code String} of comma-separated file-extensions for 
+	 * resources not to be processed
+	 * 
+	 * @param extensionsToMiss file-extensions string
+	 * @throws IllegalArgumentException if {@code extensionsToMiss} is
+	 * {@code null}
+	 */
+	public void setExtensionsToMiss(String extensionsToMiss) {
+		if (extensionsToMiss == null) {			
+			throw new IllegalArgumentException("extensionsToMiss is null");
+		} 
+		
+		this.extensionsToMiss = extensionsToMiss;
+		getConfig().setProperty(EXTENSIONS_TO_MISS, extensionsToMiss);
+	}
+
+	/**
+	 * 
+	 * @return {@code Vector} of file extensions that are ignored.
+	 * By default elements of the returned vector are
+	 * {@code "jpg", "gif", "jpeg", "ico", "tiff", "png", "bmp"}
+	 *  
+	 */
+	public Vector<String> getExtensionsToMissVector() {
+		if (extensionsToMiss.trim().equals(EMPTY_STRING)) {
+	    	return new Vector<>();
+	    }
+		
+		String[] tempArray = extensionsToMiss.replaceAll(WHITESPACE, 
+	    		EMPTY_STRING).split(COMMA);
+		
+		Set<String> tempSet = new HashSet<>(Arrays.asList(tempArray));
+		tempSet.remove(EMPTY_STRING);
+		return new Vector<>(tempSet); 
+	}
+
+	public String getFailCaseString() {
+		return failCaseString;
+	}
+
+	public void setFailCaseString(String failCaseString) {
+		if (failCaseString == null) {			
+			throw new IllegalArgumentException("failCaseString is null");
+		}
+		if (!failCaseString.isEmpty()) {

```

Comment:
If the empty string is not acceptable we should throw an exception instead (like above).
Also, this should be validated in `OptionsBruteForcePanel` to inform the user that it should not be left empty.